lib/remoteproc/elf_loader.c

```diff
@@ -184,33 +184,32 @@ static void elf_parse_segment(void *elf_info, const void *elf_phdr,
 		const Elf32_Phdr *phdr = elf_phdr;
 
 		if (p_type != NULL)
-			*p_type = (unsigned int)phdr->p_type;
+			*p_type = phdr->p_type;
 		if (p_offset != NULL)
-			*p_offset = (size_t)phdr->p_offset;
+			*p_offset = phdr->p_offset;
```

Comment:
The casting is required as the program header is using 32/bit type, but the size_t, metal_phys_addr_t is system dependent. it maynot be the same as the one defined in the program header structure.

Reply:
Fix